### PR TITLE
use libfuzzer to fuzz the iked-parser

### PR DESCRIPTION
**Pull request by linderd**
instead of fuzz.c and the test_helper-functions the parser is fuzzed by the clang-bundled libfuzzer.
also added a dictionary to preserve the functionality of the old fuzz-test and get faster fuzzing. 
in run_test.sh all parameters are given to start the test. 
for compiling clang must be used with -fsanitize=fuzzer and the compiled iked-functions must be linked (like before).

### CODE REVIEW

**Review by tobhe — CHANGES_REQUESTED** [review]
Thank you, this looks pretty promising!
Little nit: I think it would be better keep the old parser test around for now and move the new test to `regress/parser-libfuzzer`.
This would allow us to activate the test only for linux for now, which should fix the macOS CI failure.

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -53,10 +53,10 @@ jobs:
       shell: bash
       run: ./dhtest
 
-    - name: Test Parser
-      working-directory: ${{github.workspace}}/build/regress/parser
+    - name: Test Libfuzz-Parser
+      working-directory: ${{github.workspace}}/build/regress/parser-libfuzzer
       shell: bash
-      run: ./test_parser
+      run: bash run_test.sh
```
Comment (tobhe) [inline .github/workflows/cmake.yml:59]: Looks like the CI is complaining with this test activated. Maybe we should keep the old `test_parser` for the CI and upgrade to libfuzz once we fixed the low hanging fruits.

**Review by tobhe — CHANGES_REQUESTED** [review]

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (linderd) [reply .github/workflows/cmake.yml:59]: fixed it with a workaround: libfuzzer will return 0 on found bugs, so CI doesn't fail.

**Review by tobhe — APPROVED** [review]

**Comment by tobhe** [comment]
Thank you, I just squashed and merged your change :+1:

**Pull request closed without merging**